src/Equinor.ProCoSys.IPO.Command/Email/SmtpService.cs

```diff
@@ -0,0 +1,115 @@
+        {
+            // SendAsync is not thread-safe, so to ensure it is not busy sending an mail I instantiate an new SmtpClient
+            var client = new System.Net.Mail.SmtpClient(_smtp.Host, _smtp.Port);
+            client.EnableSsl = _smtp.EnableSSL;
+            client.Credentials = new NetworkCredential(_smtp.Email, _smtp.Password);
+            client.SendAsync(message, token);
+        }
+
+        private String CreateInviteString(MailMessage message, Invitation invitation, string projectName)
+        {
+            StringBuilder str = new StringBuilder();
+            str.AppendLine("BEGIN:VCALENDAR");
+            str.AppendLine("PRODID:-//Equinor//ProCoSys//EN");
+            str.AppendLine("VERSION:2.0");
+            str.AppendLine("METHOD:REQUEST");
+            str.AppendLine("X-MS-OLK-FORCEINSPECTOROPEN:TRUE");
+            str.AppendLine("BEGIN:VEVENT");
+            str.AppendLine(string.Format("DTSTART:{0:yyyyMMddTHHmmssZ}", invitation.StartTimeUtc));
+            str.AppendLine(string.Format("DTSTAMP:{0:yyyyMMddTHHmmssZ}", DateTime.UtcNow));
+            str.AppendLine(string.Format("DTEND:{0:yyyyMMddTHHmmssZ}", invitation.EndTimeUtc));
+            str.AppendLine("LOCATION: " + invitation.Location);
+            str.AppendLine(string.Format("UID:{0}", Guid.NewGuid()));
+            str.AppendLine(string.Format("DESCRIPTION:{0}", message.Body));
+            str.AppendLine(string.Format("X-ALT-DESC;FMTTYPE=text/html:{0}", message.Body));
+            str.AppendLine(string.Format("SUMMARY:{0}", message.Subject));
+            str.AppendLine(string.Format("ORGANIZER;CN=\"{0}\":mailto:{1}", message.From.DisplayName, message.From.Address));
+            message.To.ToList().ForEach(y =>
+            {
+                str.AppendLine(string.Format("ATTENDEE;CN=\"{0}\";RSVP=TRUE:mailto:{1}", y.DisplayName, y.Address));
+            });
+            str.AppendLine("CLASS:PUBLIC");
+            str.AppendLine("TRANSP:OPAQUE");
+
+            str.AppendLine("BEGIN:VALARM");
+            str.AppendLine("TRIGGER:-PT15M");
+            str.AppendLine("ACTION:DISPLAY");
+            str.AppendLine("DESCRIPTION:Reminder");
+            str.AppendLine("END:VALARM");
+            str.AppendLine("END:VEVENT");
+            str.AppendLine("END:VCALENDAR");
+            return str.ToString();
+        }
+
+        private Attachment CreateInviteAttachment(String inviteString)
+        {
+            byte[] byteArray = Encoding.ASCII.GetBytes(inviteString);
+            MemoryStream stream = new MemoryStream(byteArray);
+
+            Attachment attachment = new Attachment(stream, "invite.ics");
+            attachment.ContentType = new ContentType("text/calendar");
+
+            attachment.TransferEncoding = TransferEncoding.QuotedPrintable;
+            return attachment;
+        }
+
+        public async Task SendSmtpWithInviteAsync(Invitation invitation, string projectName, Person organizer, string pcsBaseUrl)
```

Comment:
This is not an aync method ... no awaits. 
make it Void and rename to SendSmtpWithInvite